socorro/collector/submitter_app.py

```diff
@@ -313,40 +263,52 @@ def source_iterator(self):
 
     #--------------------------------------------------------------------------
     def _infinite_iterator(self):
+        """the default iterators will yield None objects if there is nothing
+        in the queue.  The submitter doesn't want those, so it declines to
+        to yield them"""
         while True:
-            for crash_id in self.new_crash_source.new_crashes():
-                yield crash_id
-
-    #--------------------------------------------------------------------------
-    def _all_iterator(self):
-        for crash_id in self.new_crash_source.new_crashes():
-            yield crash_id
+            for crash_id in self._basic_iterator():
+                if crash_id is not None:
+                    yield crash_id
```

Comment:
Or...
```
for crash_id in super(SubmitterApp, self)._infinite_iterator():
    if crash_id is not None:
        yield crash_id
```
That way, this overrider doesn't need to know it's `self._basic_iterator`.